workflow/executor.go

```diff
@@ -71,6 +72,10 @@ type ExecutableWorkflow interface {
 
 	// OutputSchema returns the schema for the possible outputs of this workflow.
 	OutputSchema() map[string]*schema.StepOutputSchema
+
+	// Namespaces returns a namespaced collection of objects for the inputs
+	// and outputs of each stage in the step's lifecycles.
```

Comment:
I still feel that a more explicit schema description would be useful here: what's the first level map key? what's the second level map key? We're mapping something to something to `ObjectSchema`, but it seems that knowing that each of those two string keys *mean* would be fairly critical to anyone reading the interface comment.